### PR TITLE
Core/FifoAnalyzer: Tidy up interface

### PR DESCRIPTION
**Pull request by lioncash**
Tidies up the interface by hiding functions that don't need to be exposed, making them internally linked. While we're at it, we can convert the `DecodeMode` enum over to be an enum class and make use of `std::array` to help with dehardcoding sizes (and prevent implicit array->pointer decay).

The diff is larger than usual for this change due to one function needing to be relocated once it was made internally linked. This is part of the first commit, so I'd suggest reviewing the final two commits by themselves.

### CODE REVIEW

File: Source/Core/Core/FifoPlayer/FifoAnalyzer.cpp [hunk 1]
```diff
@@ -4,10 +4,10 @@
 
 #include "Core/FifoPlayer/FifoAnalyzer.h"
 
-#include <algorithm>
```
Comment (BhaaLseN) [inline Source/Core/Core/FifoPlayer/FifoAnalyzer.cpp]: Why did you remove this `#include`? `AnalyzeCommand` still uses `std::accumulate` which comes from here.

Comment (lioncash) [reply Source/Core/Core/FifoPlayer/FifoAnalyzer.cpp]: `std::accumulate` is located within `<numeric>`

Comment (BhaaLseN) [reply Source/Core/Core/FifoPlayer/FifoAnalyzer.cpp]: Got carried away by the [cppreference.com URL](https://en.cppreference.com/w/cpp/algorithm/accumulate), which has `algorithm` in it (for the algorithm library, not the `algorithm` header, as it turns out...)
